cloud/amazon/ec2_asg.py

```diff
@@ -503,33 +506,24 @@ def create_autoscaling_group(connection, module):
             module.fail_json(msg=str(e))
         return(changed, asg_properties)
 
-
 def delete_autoscaling_group(connection, module):
     group_name = module.params.get('name')
     groups = connection.get_all_groups(names=[group_name])
     if groups:
         group = groups[0]
-        group.max_size = 0
-        group.min_size = 0
-        group.desired_capacity = 0
-        group.update()
-        instances = True
-        while instances:
-            tmp_groups = connection.get_all_groups(names=[group_name])
-            if tmp_groups:
-                tmp_group = tmp_groups[0]
-                if not tmp_group.instances:
-                    instances = False
-            time.sleep(10)
-
-        group.delete()
-        while len(connection.get_all_groups(names=[group_name])):
-            time.sleep(5)
-        changed=True
-        return changed
+        group.shutdown_instances()
+        for delete_attempts in range(12):
+            try:
+                group.delete()
+                return True
+            except BotoServerError, err:
+                if err.error_code == 'ScalingActivityInProgress' or err.error_code == 'ResourceInUse':
+                    time.sleep(min(random.random() * 2 ** delete_attempts, 300))
```

Comment:
Seems like this process could potentially wait for a very long time before returning, this should be an option.